src/Strategy/RPM.hs

```diff
@@ -0,0 +1,132 @@
+    BuildRequires RPMDependency
+    deriving (Eq, Ord, Show, Generic)
+
+newtype Dependencies = Dependencies
+    { depBuildRequires :: [RPMDependency] 
+    } deriving (Eq, Ord, Show, Generic)
+{-
+Yep...  Those ones right above me.  Kinda rude.  They're actually standing on my head.
+-}
+
+discover :: HasDiscover sig m => Path Abs Dir -> m ()
+discover = walk $ \dir _ files ->
+  case find (\f -> ".spec" `isSuffixOf` fileName f) files of
+    Nothing -> pure WalkContinue
+    Just specFile -> do
+      runSimpleStrategy "rpm-spec" RustGroup $ fmap (mkProjectClosure dir) (analyze specFile)
+      pure WalkSkipAll
+
+analyze :: (Has ReadFS sig m, Has (Error ReadFSErr) sig m) => Path Rel File -> m (Graphing Dependency)
+analyze specFile = do
+  specFileText <- readContentsText specFile
+  pure . buildGraph $ getSpecDeps specFileText
+
+mkProjectClosure :: Path Rel Dir ->  Graphing Dependency -> ProjectClosureBody
+mkProjectClosure dir graph = ProjectClosureBody
+  { bodyModuleDir    = dir
+  , bodyDependencies = dependencies
+  , bodyLicenses     = []
+  }
+  where
+  dependencies = ProjectDependencies
+    { dependenciesGraph    = graph
+    , dependenciesOptimal  = NotOptimal
+    , dependenciesComplete = NotComplete
+    }
+
+toDependency :: RPMDependency -> Set SpecFileLabel -> Dependency
+toDependency pkg = foldr applyLabel Dependency 
+  { dependencyType = CargoType 
+  , dependencyName = rpmDepName pkg
+  , dependencyVersion = rpmConstraint pkg
+  , dependencyLocations = []
+  , dependencyEnvironments = []
+  , dependencyTags = M.empty
+  }
+  where
+    applyLabel :: SpecFileLabel -> Dependency -> Dependency
+    applyLabel (RequiresType env) dep = dep { dependencyEnvironments = env : dependencyEnvironments dep }
+
+buildGraph :: Dependencies -> Graphing Dependency
+buildGraph Dependencies{..} = run . withLabeling toDependency $ do
+  traverse_ direct depBuildRequires
+  traverse (flip label $ RequiresType EnvDevelopment) depBuildRequires
+
+-- buildConstraint tail = RPMConstraint version <$> comparator
+buildConstraint :: Text -> Maybe VerConstraint
+buildConstraint tail = constraint
+  where
+    (comparatorStr, rawVersion) = splitOnceOn " " $ T.strip tail
+    version = T.strip rawVersion
+    constraint = case T.strip comparatorStr of
+      "<=" -> Just $ CLessOrEq version
+      "<" -> Just $ CLess version
+      ">=" -> Just $ CGreaterOrEq version
+      ">" -> Just $ CGreater version
+      "=" -> Just $ CEq version
+      "==" -> Just $ CEq version
+      _ -> Nothing
+
+splitOnceOn :: Text -> Text -> (Text, Text)
+splitOnceOn needle haystack = (head, strippedTail)
+  where
+    len = T.length needle
+    (head, tail) = T.breakOn needle haystack
+    strippedTail = T.drop len tail
+
+getTypeFromLine :: Text -> Maybe RequiresType
+getTypeFromLine line = safeReq
+  where
+    (header, value) = splitOnceOn ": " line
+    (pkgName, rawConstraint) = splitOnceOn " " $ T.strip value
+    isSafeName name = not $ "%{" `T.isInfixOf` name
```

Comment:
[nit] I think I would prefer either not defining `isSafeName` and inlining the logic or adding a type signature for the function.

Reply:
I disagree here.  `isSafeName` takes a name and tells you if it's safe.  My instinct here would assume a `Text -> Bool`, which is exactly what it is (if that wasn't it, I would assume a newtype wrapper around a `Text` or other `String`-like).  Inlining decreases readability for sure, and I think a type sig is pretty unnecessary.

Reply:
Type signatures on all of the non-`splitOnceOn` bindings would be nice. It took me a while to mentally map the types (especially on `req`)

Reply:
If the other bindings need it, then consistency dictates that `isSafeName` should also get it.